pages/api/pointsOfInterest.ts

```diff
@@ -1,14 +1,37 @@
-export default async function handler(req: any, res: any){
+import { locationLabels } from "@/constants/constants"
+import { json } from "stream/consumers"
+
+
+export default async function handler(req: any, res: any) {
     const body = JSON.parse(req.body)
     let city = body['city']
     let pointOfInterest = body['point']
 
-    console.log(req.body)
-    console.log(city)
-    console.log(pointOfInterest)
-    const placesApi = `https://maps.googleapis.com/maps/api/place/textsearch/json?query=${pointOfInterest}{+in+${city}&key=${process.env.NEXT_PUBLIC_GOOGLE_MAPS_API_KEY}`
-    const results = await fetch(placesApi)
+    //creating an array of promises 
+    let promiseArr = []
+    for (let keyword of locationLabels[pointOfInterest]) {
+        promiseArr.push(fetch(`https://maps.googleapis.com/maps/api/place/textsearch/json?query=${keyword}{+in+${city}&key=${process.env.NEXT_PUBLIC_GOOGLE_MAPS_API_KEY}`))
+    }
+    // assigning the now fullfilled promises to a variable
+    const googleResults = await Promise.all(promiseArr)
+
+    //makes the raw data into json 
+    let jsonPromises = []
+    for (let promise of googleResults) {
```

Comment:
[sand] Using "promise" as a variable in this loop is a bit misleading because you are looping through resolved promises (ie. no longer promises)
A better name would be obj or something

The promise here will be using the _.json()_ on that object. THAT returns a promise

Reply:
Changed it to obj